src/components/Input/Input.tsx

```diff
@@ -1,20 +1,67 @@
-import { ChangeEvent } from "react";
+import {
+  ChangeEvent,
+  useState,
+  useEffect,
+  Dispatch,
+  SetStateAction,
+} from "react";
 import "./Input.css";
+import { v4 as uuidv4 } from "uuid";
+import { formattedDate } from "../DateAndDay/DateAndDay";
+import { TaskProps } from "../TaskList/TaskList";
 
 interface InputProps {
-  valid: boolean;
-  touched: boolean;
-  enteredTask: string;
-  checkValidity: (e: ChangeEvent<HTMLInputElement>) => void;
+  setTasks: Dispatch<SetStateAction<TaskProps[]>>;
+  toggleHandler: (arg: boolean) => void;
 }
 
-function Input({ valid, touched, enteredTask, checkValidity }: InputProps) {
+function Input({ setTasks, toggleHandler }: InputProps) {
+  const [valid, setValid] = useState(false);
+  const [enteredTask, setEnteredTask] = useState("");
+  const [touched, setTouched] = useState(false);
   const error = !valid && touched && (
     <div className="error">
       <p>Enter Something</p>
     </div>
   );
 
+  function checkValidity(e: ChangeEvent<HTMLInputElement>) {
+    setEnteredTask(e.target.value);
+  }
+
+  useEffect(() => {
+    const handleKey = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        toggleHandler(false);
+      }
+      if (event.key === "Enter") {
```

Comment:
`<form>` tag automatically enables pressing enter to submit form.
Remove this condition and put it into onSubmit event.